src/test/java/com/google/api/generator/gapic/composer/ServiceClientSampleCodeComposerTest.java

```diff
@@ -0,0 +1,452 @@
+import com.google.api.generator.engine.ast.VaporReference;
+import com.google.api.generator.gapic.composer.samplecode.SampleCodeWriter;
+import com.google.api.generator.gapic.model.Field;
+import com.google.api.generator.gapic.model.Method;
+import com.google.api.generator.gapic.model.MethodArgument;
+import com.google.api.generator.gapic.model.ResourceName;
+import com.google.api.generator.gapic.model.ResourceReference;
+import com.google.api.generator.gapic.protoparser.Parser;
+import com.google.protobuf.Descriptors.FileDescriptor;
+import com.google.showcase.v1beta1.EchoOuterClass;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ServiceClientSampleCodeComposerTest {
+  private static final String PACKAGE_NAME = "com.google.showcase.v1beta1";
+  private static final TypeNode clientType =
+      TypeNode.withReference(
+          VaporReference.builder().setName("EchoClient").setPakkage(PACKAGE_NAME).build());
+
+  FileDescriptor echoFileDescriptor;
+  Map<String, ResourceName> resourceNames;
+
+  @Before
+  public void setUp() {
+    echoFileDescriptor = EchoOuterClass.getDescriptor();
+    resourceNames = Parser.parseResourceNames(echoFileDescriptor);
+  }
+
+  @Test
+  public void composeUnaryRpcMethodSampleCode_resourceNameMethodArgument() {
+    TypeNode inputType =
+        TypeNode.withReference(
+            VaporReference.builder().setName("EchoRequest").setPakkage(PACKAGE_NAME).build());
+    TypeNode outputType =
+        TypeNode.withReference(
+            VaporReference.builder().setName("EchoResponse").setPakkage(PACKAGE_NAME).build());
+    TypeNode resourceNameType =
+        TypeNode.withReference(
+            ConcreteReference.withClazz(com.google.api.resourcenames.ResourceName.class));
+    MethodArgument arg =
+        MethodArgument.builder()
+            .setName("parent")
+            .setType(resourceNameType)
+            .setField(
+                Field.builder()
+                    .setName("parent")
+                    .setType(TypeNode.STRING)
+                    .setResourceReference(
+                        ResourceReference.withType("showcase.googleapis.com/AnythingGoes"))
+                    .build())
+            .setIsResourceNameHelper(true)
+            .build();
+    List<List<MethodArgument>> signatures = Arrays.asList(Arrays.asList(arg));
+    Method unaryMethod =
+        Method.builder()
+            .setName("echo")
+            .setMethodSignatures(signatures)
+            .setInputType(inputType)
+            .setOutputType(outputType)
+            .build();
+    String results =
+        SampleCodeWriter.write(
+            ServiceClientSampleCodeComposer.composeUnaryRpcMethodSampleCode(
+                unaryMethod, signatures.get(0), clientType, resourceNames));
+    String expected =
+        "try (EchoClient echoClient = EchoClient.create()) {\n"
```

Comment:
Instead of the newline chars, could we please use [`LineFormatter.lines()`](https://github.com/googleapis/gapic-generator-java/blob/master/src/test/java/com/google/api/generator/testutils/LineFormatter.java#L18)? Same elsewhere in this file; example usages are in other tests.

Reply:
Thanks for pointing out. Updated the code.